flashgbx: init at 3.37

grahamnorris opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This PR adds [FlashGBX](https://github.com/lesserkuma/FlashGBX), a command-line and GUI utility for reading and writing to Game Boy, Game Boy Color, and Game Boy Advance cartridges using [GBxCart RW](https://www.gbxcart.com/) hardware.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Tested that this launches on aarch64-darwin and launches and functions correctly on x86_64-linux, including wayland.
The desktop icon does not appear correct on KDE plasma wayland, but AFAICT, this is due to the presence of a PNG rather than SVG icon.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

grahamnorris committed:
flashgbx: init at 3.37

grahamnorris committed:
maintainers: add grahamnorris

lolbinarycat commented on pkgs/by-name/fl/flashgbx/package.nix:
```diff
@@ -0,0 +1,68 @@
+    };
+  in builtins.map makeDesktopItem [
+    common
+    (common // {
+      name = "${common.name}";
+      exec = "${common.exec}";
+    })
```
it's unclear to me what exactly this is doing.  as far as i can tell it is overriding `common.name` with `common.name`, which seems like it wouldn't do anything?

grahamnorris replied:
Thanks for pointing this out. This was a remnant from the example I started from. I have now looked at a few other examples of `makeDesktopItem` and adapted the code, removing the mapping.

lolbinarycat commented on pkgs/by-name/fl/flashgbx/package.nix:
```diff
@@ -0,0 +1,62 @@
+  src = fetchFromGitHub {
+    repo = "FlashGBX";
+    owner = "lesserkuma";
+    rev = "${version}";
```
```suggestion
    rev = version;
```

there's no need to interpolate here

grahamnorris replied:
Another good catch, thanks.

lolbinarycat commented on pkgs/by-name/fl/flashgbx/package.nix:
```diff
@@ -0,0 +1,62 @@
+    repo = "FlashGBX";
+    owner = "lesserkuma";
+    rev = version;
+    sha256 = "sha256-MbeQKdmAGt4bTSleU1Dlve7fLDfldS2gIYUCIbgDAk4=";
```
```suggestion
    hash = "sha256-MbeQKdmAGt4bTSleU1Dlve7fLDfldS2gIYUCIbgDAk4=";
```

this is a small nitpick, but i'm pretty sure the `hash` attribute is preferred these days.

grahamnorris replied:
Okay, I'm happy to use the preferred nomenclature.

rafameou commented:
Result of `nixpkgs-review pr 291763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashgbx</li>
    <li>flashgbx.dist</li>
  </ul>
</details>

rafameou commented on pkgs/by-name/fl/flashgbx/package.nix:
```diff
@@ -0,0 +1,62 @@
+      install -D FlashGBX/res/icon.png $out/share/icons/hicolor/256x256/apps/flashgbx.png
+  '';
+
+  format = "pyproject";
```
```suggestion
  pyproject = true;
```
https://github.com/NixOS/nixpkgs/issues/253154#issuecomment-1712673789

grahamnorris replied:
Okay, thanks for this and the reference. I have addressed this in the latest push.

rafameou commented:
I would also suggest changing the title of the pull request to "flashgbx: init at <version>"

rafameou commented:
Result of `nixpkgs-review pr 291763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashgbx</li>
    <li>flashgbx.dist</li>
  </ul>
</details>

rafameou reviewed (approved):
LGTM, I can't test completely since I don't have the hardware, but overall it seems to be working well.

pbsds commented:
Result of `nixpkgs-review pr 291763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashgbx</li>
    <li>flashgbx.dist</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [ ] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored


please add `setuptools` to nativeBuildInputs, and please swap the order of the commits

grahamnorris commented:
Thanks for the review. I have addressed the comments.

I am not fully familiar with the nuances between the different `*Inputs` for a nix packages, but without `setuptools` also in `propagatedBuildInputs`, I only got the CLI version of the application instead of the GUI version on aarch64-darwin, so it seems the package has a run-time dependency on `setuptools`.

pbsds commented on pkgs/by-name/fl/flashgbx/package.nix:
```diff
@@ -0,0 +1,62 @@
+  pyproject = true;
+
+  nativeBuildInputs = [
+    python3Packages.setuptools
```
```suggestion
    python3Packages.setuptools
    copyDesktopItems
```

pbsds commented:
the difference between nativeBuildInputs and buildInputs only matters for cross builds, where e.g. a x86 machine builds a ARM package. nativeBuildInputs would then be the x86 versions which can run on the host during build, while buildInputs would be ARM packages that get linked and propagated into the final build artifact. nativeBuildInputs get added to PATH, while buildInputs are added to configure scripts and such.
